xtask/codegen/css.ungram

```diff
@@ -62,17 +63,43 @@ CssRule =
 
 // .header, .app {}
 // ^^^^^^^^^^^^^
-CssSelectorList = (AnyCssSelectorPattern (',' AnyCssSelectorPattern)*)
-
-AnyCssSelectorPattern =
-	CssIdSelectorPattern
-	| CssClassSelectorPattern
-	| CssUniversalSelectorPattern
-	| CssAttributeSelectorPattern
-	| CssPseudoClassSelectorPattern
-	| CssTypeSelectorPattern
-	| CssCombinatorSelectorPattern
-	| CssBogusPattern
+CssSelectorList = (AnyCssSelector (',' AnyCssSelector)*)
+
+AnyCssSelector =
+	CssComplexSelector
+	| CssCompoundSelector
+	| CssBogusSelector
+
+// div a {}
+// ^^^^^
+// div > a {}
+// ^^^^^^^
+CssComplexSelector =
+	left: AnyCssSelector
+	combinator: CssComplexSelectorCombinator
+	right: AnyCssSelector
+
+CssComplexSelectorCombinator =
```

Comment:
I've been thinking about it.
I'm not sure that the optional combinator could solve the problem.

I'd like to try to implement `lookahead` and `CssLexContext` for `CssLexer`.
We have the same approach for the js parser:
https://github.com/biomejs/biome/blob/d53b1dbdfe2e89e47fafd9f6bc51542526eeee8f/crates/biome_js_parser/src/lexer/mod.rs#L60-L81

It can help us to understand when we need to lex a space as a combinator or as a trivia.

Reply:
Sounds good! Thank you